src/libs/ajax/ajax-common.ts

```diff
@@ -35,6 +35,29 @@ export const withRetryOnError = _.curry((shouldNotRetryFn, wrappedFetch) => asyn
   return wrappedFetch(...args);
 });
 
+const TIMEOUT_DURATION = 10000;
+
+export function makeRequestRetry(request: Function, retriesLeft: number) {
+  return Promise.race([
+    request(),
+    new Promise((_, reject) => setTimeout(() => reject(new Error('Request timed out')), TIMEOUT_DURATION)),
+  ])
+    .then((response) => {
+      if (response.ok) {
+        return response.json();
+      }
+      throw new Error('Network response was not ok.');
+    })
+    .catch((error) => {
+      if (retriesLeft === 0) {
+        throw error;
+      }
+      return new Promise((resolve) => {
+        setTimeout(resolve, TIMEOUT_DURATION);
+      }).then(() => makeRequestRetry(request, retriesLeft - 1));
```

Comment:
I really don't like doing this recursively... does the timeout even work like this, it seems like we reset it each look until our number of tries are expended.

I'd really like to do it declaratively in a while loop like this: https://github.com/DataBiosphere/terra-ui/blob/dev/integration-tests/utils/integration-helpers.js#L64

something like

```
let numTries = 5
const startTime = Date.getCurrentTimeInMillis()
const TIMEOUT_DURATION = ...


while (numTries > 0 && !((currTime - startTime) > TIMEOUT_DURATION) {
...
}
```

Reply:
So, there is a different function that does retrys, I just didn't like the datetime calculation part. I'm happy to just use that function instead.

But the function does work as intended, and the retry happens within 10 seconds instead of the minute or so it takes to get the 504 without the wrapper.

Reply:
Yeah, I'd like to use the existing one if it can at all fit our use case... I find retrying is always one of those things that no matter how hard you think about it you will always find errors down the line with it. You can always parameterize the existing one the better handle the datetime like you want.

Reply:
How do I even use withRetryOnError correctly in this context?

Reply:
I second Justin's sentiment that we should avoid multiple retry functions... Also, where are the unit tests?  I checked current code coverage and even withRetryOnError has no coverage.... I would much prefer we kill two birds with one stone, improve/parameterize withRetryOnError as needed... even remove/simplify currying if current usage allows... and hit it with unit tests.

Reply:
I would be very happy to remove the currying portion.

As for implementation for the retry, do you have a preference? Keep it similar to the original implementation, or continue to use my new implementation?

Reply:
My main push is to have either outcome have the retry logic well unit tested.  (and I concur with the feedback fir better types on promise return, etc.) I would gut-check your new logic recommendation with the last dev who touched this topic... I did a quick look and it appears the existing retry fn is only used on one place...

Thanks for braving into this capability and trying to make it better!

Reply:
I think I'd like to keep the PR as-is since this is a time sensitive ticket.
@msilva-broad Did you get a chance to look at my changes from this AM?